client/commands/v2/pysa_server.py

```diff
@@ -129,60 +219,102 @@ async def process_did_save_request(
             raise json_rpc.InvalidRequestError(
                 f"Document URI is not a file: {parameters.text_document.uri}"
             )
+        await self.update_errors(document_path)
+
+    async def process_copy_model_request(
+        self, parameters: lsp.DidCopyModelParameters
+    ) -> str:
+        document_path = parameters.path
+        if document_path is None:
+            raise json_rpc.InvalidRequestError(
+                f"Document path is not a file: {parameters.path}"
```

Comment:
Nit: That's not what you're checking. You're just checking that `document_path` is not `None` or `""`. I'd suggest tweaking the error (or going further an checking that `document_path` does indeed point to a file, so this error message is true)